src/subprocess.rs

```diff
@@ -94,10 +171,43 @@ impl SubProcessIntf for SubProcess {
             env.unwrap_or(DEFAULT_ENV_NAME).to_string(),
         );
 
-        // Add in the items from the CloudTruth environment, and overrides.
-        self.env_vars
-            .extend(self.get_ct_vars(org_id, env, environments)?);
-        self.env_vars.extend(self.process_overrides(overrides));
+        // Add in the items from the CloudTruth environment (looking for collisions)
+        let mut collisions: Vec<String> = vec![];
+        let ct_vars = self.get_ct_vars(org_id, env, environments)?;
+        for (key, value) in ct_vars {
+            if !self.env_vars.contains_key(&key) {
+                // when not already, insert it
+                self.env_vars.entry(key).or_insert(value);
+            } else {
+                let orig = self.env_vars.get(&key).unwrap();
+                if inherit == Inheritance::Exclusive && value != *orig {
+                    collisions.push(key);
+                } else if inherit == Inheritance::Overlay {
+                    self.env_vars.insert(key, value);
+                }
+                // if doing Underlay, the local environment value is already set
+            }
+        }
+
+        // Add in the items from the overrides (looking for collisions)
+        let over_vars = self.process_overrides(overrides);
+        for (key, value) in over_vars {
+            let orig = self.env_vars.get(&key).unwrap();
+            if inherit == Inheritance::Exclusive
+                && self.env_vars.contains_key(&key)
+                && value != *orig
+            {
+                collisions.push(key);
+            } else {
+                // use the "set" value as the final answer, when not worrying about collisions
+                self.env_vars.insert(key, value);
+            }
+        }
+
+        // return the error(s) if there were not any collisions
+        if !collisions.is_empty() {
```

Comment:
Structure this however you'd like, but I think the more idiomatic approach here is to avoid the explicit `return`:

```rust
if !collisions.is_empty() {
    Err(SubProcessError::EnvironmentCollisions(collisions));
} else {
    // Remove the specified values.
    for r in removals {
        self.env_vars.remove(r.as_str());
    }

    Ok(())
}
```

Reply:
Thanks for the tip... Appreciate the help with the new language.